src/TabbedWebView.vala

```diff
@@ -18,26 +18,38 @@
 *
 */
 
-public class TabbedWebView : Granite.Widgets.Tab {
-    public WebKit.WebView web_view;
+public class TabbedWebView : WebKit.WebView {
+    public bool load_cookies { get; construct; }
     public CertButton.Security security { get; private set; }
 
-    public TabbedWebView (string uri, bool load_cookies) {
-        web_view = new WebKit.WebView ();
+    public TabbedWebView (bool load_cookies) {
+        Object (load_cookies: load_cookies);
+    }
 
-        page = web_view;
+    construct {
+        if (load_cookies) {
+            var cookies_db_path = Path.build_path (
+                Path.DIR_SEPARATOR_S,
+                Environment.get_user_config_dir (),
+                "epiphany",
+                "cookies.sqlite"
+            );
 
-        setup_web_view (load_cookies);
+            if (FileUtils.test (cookies_db_path, FileTest.IS_REGULAR)) {
+                var cookie_manager = get_context ().get_cookie_manager ();
 
-        web_view.insecure_content_detected.connect (() => {
-            security = CertButton.Security.MIXED_CONTENT;
-        });
+                cookie_manager.set_accept_policy (WebKit.CookieAcceptPolicy.ALWAYS);
+                cookie_manager.set_persistent_storage (cookies_db_path, WebKit.CookiePersistentStorage.SQLITE);
+            } else {
+                critical ("No cookies store found, not saving the cookies…");
```

Comment:
Is this being a critical now intentional?

Reply:
Yeah, I don't think this does anything anymore since we started to ship Epiphany in Flatpak, so I think it's good for it to make noise so we don't forget that it's broken :sweat_smile: I'm not sure if the ultimate solution is to remove this or what but I can revert that if you'd like :)